test/ecto/query/planner_test.exs

```diff
@@ -839,6 +839,16 @@ defmodule Ecto.Query.PlannerTest do
     assert Macro.to_string(hd(hd(query.joins).source.query.wheres).expr) =~ "in %Ecto.SubQuery{"
   end
 
+  test "normalize: parent_as used at right side term of union expression" do
```

Comment:
it is not clear to me what kind of tests are necessary to test this change. Please advise.